deploy/namespace.yaml

```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: operators
```

Comment:
Maybe changing to `kogito-operator-system` to align with the upcoming SDK 1.0 migration? Is this namespace an SDK 0.x recommendation?

Reply:
I chose the namespace `operators` to be in coherence with the global-operators namespace when we install OLM in k8s. I am not sure but using yaml the installation would be done in k8s without olm installed, and by default in cli we are deploying the operator in cluster-scope, and to verify if it exists we check if the `kogito-operator` deployment is present in `global-operator-namespace` depending on the env(i.e.: OCP: `openshift-operators`, K8s: `operators`)

Reply:
> I chose the namespace operators to be in coherence with the global-operators namespace when we install OLM in k8s. 

Makes sense, but when installing without `olm`, as an user I might want to see my operator installed in its namespace instead of a global namespace of any kind, since without OLM that won't make sense whatsoever.

To summarize:

1. On environments with OLM, ~we~ (OLM) will install in the global namespace, no matter the name
2. On environments without OLM, on a cluster-scoped installation, the namespace should follow kubebuilder suggestion: `{operator-name}-system`
3. For namespace scoped installation, doesn't matter.

Reply:
Changed